src/CoolTime.cpp

```diff
@@ -857,3 +813,103 @@ String CoolTime::formatDigits(int digits) {
 
   return (String(digits));
 }
+
+/**
+ *  CoolTime::timePooConfigl()
+ *
+ *  utility method for chosing the server with the best ping
+ *  returns 0 if it fails or returns the number of the const char* timeServer[]
+ *
+ *  \return formatted string of the input digit
+ */
+int CoolTime::timePoolConfig() {
+
+#if DEBUG == 1
+  Serial.println();
+  Serial.println(F("Enter timePoolConfig"));
+  Serial.println();
+#endif
+
+  Serial.println(F("Please wait, performing NTP server test..."));
+  unsigned long timer[6];
+  bool timeout[6];
+  for (int i = 0; i <= SERVERCOUNT; i++) {
+    timer[i] = 0;
+    timeout[i] = false;
+  }
+  int result = -1;
+  for (int j = 1; j <= NTP_OVERSAMPLE; j++) {
+  //check all servers to get the fastest
+    for (int i = 0; i <= SERVERCOUNT - 1; i++) {
+      while (Udp.parsePacket() > 0)
+        ; // discard any previously received packets
+
+      WiFi.hostByName(timeServer[i], timeServerIP);
+
+#if DEBUG == 1
+      Serial.println(timeServer[i]);
+      Serial.println(timeServerIP);
+      Serial.println(F("Transmit NTP Request"));
+#endif
+
+      sendNTPpacket(timeServerIP);
+
+      uint32_t beginWait = millis();
+
+      while (millis() - beginWait < 2200) {
+        int size = Udp.parsePacket();
+        //yield();
+        if (size >= NTP_PACKET_SIZE) {
+          timer[i] = timer[i] + (millis() - beginWait);
+
+#if DEBUG == 1
+          Serial.print(F("Receive NTP Response from "));
+          Serial.println(timeServer[i]);
+          Serial.printf("latency : %ld ms \n \n", timer[i] );
+#endif
+
+          break;
+        }
+        if (millis() - beginWait >= 2000) {
+          timeout[i] = true;
+
+#if DEBUG == 1
+          Serial.println(F("Hit Timeout !"));
+          Serial.println();
+#endif
+          break;
+        }
+      }
+      
+    }
+  }
+
+  //compare values and if we run into timeout...
+  unsigned long temp = 0;
+  //get first value 
+  if (timer[0] != 0 && !timeout[0]) {
+    temp = timer[0];
+    result = 0;
+  }
+  //compare if a other server was faster, check also if we hit timeout
+  for (int i = 1; i <= SERVERCOUNT - 1; i++) {
+    if ((timer[i] != 0) && (timer[i] < temp) && !timeout[i]){
```

Comment:
the first check `timer[i] != 0` is unnecessary, and you should check `timeout` before checking for the latency